CMakeLists.txt

```diff
@@ -90,8 +90,11 @@ set(FOLLY_SUPPORT_SHARED_LIBRARY "${BUILD_SHARED_LIBS}")
 include(FBBuildOptions)
 fb_activate_static_library_option()
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
it would be better to set `CMAKE_CXX_STANDARD` as a `CACHE` variable and remove the `if`
```
set(CMAKE_CXX_STANDARD "17" CACHE STRING  "folly C++ standard")
```